### PR TITLE
[WIP] interfaces for kube environment for bookinfo test

### PR DESCRIPTION
**Pull request by GregHanson**

### CODE REVIEW

File: tests/e2e/framework/kubernetes.go [hunk 1]
```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
...
+func (k *KubeInfo) AddApp(appYaml string) {
+	demo := App{AppYaml: util.GetResourcePath(filepath.Join(rulesDir, appYaml)), KubeInject: true}
+	k.AppManager.AddApp(&demo)
+}
+
+func (k *KubeInfo) CheckRunning() bool {
+	return util.CheckPodsRunning(k.Namespace)
+}
+
+func (k *KubeInfo) DeleteRule(rule string) error {
```
Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: nit: Might be helpful to clarify what the parameter is: `ruleYaml`? `ruleName`?  Same for the other methods below.

File: tests/e2e/framework/kubernetes.go [hunk 2]
```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 	}, nil
 }
 
+func (k *KubeInfo) GetGateway() string {
```
Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: Add docs for exposed methods?

File: tests/e2e/framework/kubernetes.go [hunk 3]
```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
...
+	return k.Ingress
+}
+
+func (k *KubeInfo) AddApp(appYaml string) {
+	demo := App{AppYaml: util.GetResourcePath(filepath.Join(rulesDir, appYaml)), KubeInject: true}
+	k.AppManager.AddApp(&demo)
+}
+
+func (k *KubeInfo) CheckRunning() bool {
```
Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: nit: Maybe `IsRunning`?

File: tests/e2e/framework/kubernetes.go [hunk 4]
```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
...
+
+func (k *KubeInfo) DeleteRule(rule string) error {
+	if err := util.KubeDelete(k.Namespace, rule); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *KubeInfo) AddRule(rule string) error {
+	if err := util.KubeApply(k.Namespace, rule); err != nil {
```
Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: Do we need to clarify that this method is actually an add or update? Maybe we can do that in the docs?

We don't seem to have an `UpdateRule` so maybe we could just rename this to `AddOrUpdateRule`?

File: tests/e2e/framework/kubernetes.go [hunk 5]
```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
...
+		log.Errorf("Kubectl apply %s failed", rule)
+		return err
+	}
+	return nil
+}
+
+func (k *KubeInfo) GenerateRule(rule, dest, user string) error {
```
Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: Can you clarify the arguments a bit more (file name? directory? fully-qualified path?, etc.)

Comment (nmittler) [inline tests/e2e/framework/kubernetes.go]: What exactly does this method do? It seems like it's just replacing `jason` with the actual user.   Two questions/comments:

1) Who is `jason`? :)
2) Is this really a common method? It seems a bit clunky being here ... perhaps we move this logic back to the call site?

File: tests/e2e/tests/bookinfo/demo_test.go [hunk 6]
```diff
@@ -91,28 +92,18 @@ func closeResponseBody(r *http.Response) {
 }
 
 func (t *testConfig) Setup() error {
-	t.gateway = "http://" + tc.Kube.Ingress
+	//t.gateway = "http://" + tc.Kube.Ingress
```
Comment (nmittler) [inline tests/e2e/tests/bookinfo/demo_test.go]: remove comment?

**Comment by costinm** [comment]
Ping, is it still active ? Please reopen if still needed.

**Pull request closed without merging**